samples/node/pub_sub/index.ts

```diff
@@ -0,0 +1,194 @@
+/* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+import { mqtt, auth, http, io, iot } from 'aws-crt';
+import { TextDecoder } from 'util';
+
+type Args = { [index: string]: any };
+
+const yargs = require('yargs');
+yargs.command('*', false, (yargs: any) => {
+    yargs
+        .option('endpoint', {
+            alias: 'e',
+            description: "Your AWS IoT custom endpoint, not including a port. "  +
+              "Ex: \"abcd123456wxyz-ats.iot.us-east-1.amazonaws.com\"",
+            type: 'string',
+            required: true  
+        })
+        .option('ca_file', {
+            alias: 'r',
+            description: 'FILE: path to a Root CA certficate file in PEM format.',
+            type: 'string',
+            required: false
+        })
+        .option('cert', {
+            alias: 'c',
+            description: 'FILE: path to a PEM encoded certificate to use with mTLS',
+            type: 'string',
+            required: false
+        })
+        .option('key', {
+            alias: 'k',
+            description: 'FILE: Path to a PEM encoded private key that matches cert.',
+            type: 'string',
+            required: false
+        })
+        .option('client_id', {
+            alias: 'C',
+            description: 'Client ID for MQTT connection.',
+            type: 'string',
+            required: true
+        })
+        .option('topic', {
+            alias: 't',
+            description: 'STRING: Targeted topic',
+            type: 'string',
+            default: 'sdk/test/NodeJSv2'
```

Comment:
Can you make this "samples/test"?
That's the exact string that python-v2 and java-v2 use. 
Being super consistent means we can share the same exact Policy, with pretty restrictive permissions, and have it work across all the v2 SDKs. We use that exact Policy in the IoT wizard in the Console. We can copy/paste the parts of the README where we advise people how to set up their policies for each sample.

dammit, java actually uses "/samples/test" with a leading forward slash
well, we tried.